src/mgmt/srv_pool.c

```diff
@@ -231,6 +231,9 @@ ds_mgmt_destroy_pool(uuid_t pool_uuid, d_rank_list_t *ranks)
 		return -DER_INVAL;
 	}
 
+	if (DAOS_FAIL_CHECK(DAOS_CHK_POOL_ORPHAN))
+		D_GOTO(out, rc = 0);
```

Comment:
Do we need this? We already have `dmg faults mgmt-svc pool $pool POOL_NONEXIST_ON_MS` which as far as I can tell does the same thing. It removes the pool entry from the MS so that the checker finds the orphaned pool.

Reply:
Good to know that. I will remove it.
I think we also need the fault injection to simulate POOL_NONEXIST_ON_ENGINE, means to destroy the pool shards on targets, but with the pool entry kept on MS.